design/ceph/ceph-telemetry.md

```diff
@@ -0,0 +1,116 @@
+# Adding Rook data to Ceph telemetry
+The solution plan agreed upon with the telemetry team is for the Rook operator to add telemetry to
+the Ceph mon `config-key` database, and Ceph will read each of those items for telemetry retrieval.
+
+## Guidelines from the Ceph telemetry team
+- Users must opt in to telemetry
+- Users must also opt into the new (delta) telemetry items added between Ceph versions
+- Rook should not add information to `config-key` keys that can grow arbitrarily large to keep space
+  usage of the mon database low
+
+## Proposed metrics for Rook to collect
+We propose naming metrics to indicate a hierarchy that can be parsed to add it to the Ceph telemetry
+collection in a more ordered fashion.
+
+For example `rook/kubernetes/version` would should be put into a JSON structure like shown:
+```json
+"rook": {
+  "kubernetes": {
+    "version"
+  }
+}
+```
+
+### Overall metrics
+- `rook-version` - Rook version.
+  - **DISCUSS:** should we add `rook/version` or rename this to maintain a consistent hierarchy?
```

Comment:
The name `rook/version` sounds right to me. IMO, the end report should have a single `rook` key that users/devs can search for, under which they can find all of the relevant information. The rook version number is a higher-level piece of information, so it could look something like this (the report gets alphabetized):

```json
"rook": {
  "csi": {
     "version"
   },
  "kubernetes": {
    "node-count",
    "version"
   },
   "version"
}
```

Reply:
+1 for `rook/version`